okhttp/src/test/java/okhttp3/CallTest.java

```diff
@@ -2812,6 +2812,43 @@ public void cancelWhileRequestHeadersAreSent_HTTP_2() throws Exception {
     assertThat(recordedRequest.getBody().readUtf8()).isEqualTo("abc");
   }
 
+  @Test
+  public void serverRespondsWithEarlyHintsHttp2() throws Exception {
+    enableProtocol(Protocol.HTTP_2);
+    server.enqueue(
+            new MockResponse()
+                    .addInformationalResponseInternal(
+                            new MockResponse().setResponseCode(103).addHeader("Link", "</style.css>; rel=preload; as=style")
+                    )
+    );
+    Request request = new Request.Builder()
+            .url(server.url("/"))
+            .post(RequestBody.create("abc", MediaType.get("text/plain")))
+            .build();
+    executeSynchronously(request)
+            .assertFailure(SocketTimeoutException.class);
```

Comment:
Showing we don't handle this well.

cc @swankjesse

Reply:
👍🏻